syntax="proto3";
package client;

import "features.proto";

enum Type{
  NOT_FOUND = 0;
  OBJECT = 1;
  HSET = 2;
  HGET = 3;
}

// 通用返回码
message CommandReply{
  int32 errcode = 1;
  string errmsg = 2;
}

message CommandHset{
    string key = 1;
    repeated string fields = 2;
    repeated tensorflow.Feature values = 3;
    repeated int64 ttls = 4;
}

message CommandHget{
  string key = 1;
  repeated string fields = 2;
}

message CommandHgetReply{
  CommandReply state = 1;
  repeated tensorflow.Feature features = 2;
}